SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialReceiver.cpp

```diff
@@ -996,19 +1021,39 @@ void USpatialReceiver::HandleUnreliableRPC(Worker_ComponentUpdateOp& Op)
 				ObjectRef.Entity = Schema_GetEntityId(EventData, SpatialConstants::UNREAL_PACKED_RPC_PAYLOAD_ENTITY_ID);
 			}
 		}
-
+		
 		UObject* TargetObject = PackageMap->GetObjectFromUnrealObjectRef(ObjectRef).Get();
 
+		// TODO(Alex): Maybe queue by ObjectRef (create jira).
+		// It may happen when:
+		// 1. Packing RPCs (so that it's sent through a different entity)
+		// 2. Object has been destroyed (to fix that we need to clean queued component udpates for removed entity in void USpatialDispatcher::ProcessOps())
 		if (!TargetObject)
 		{
-			UE_LOG(LogSpatialReceiver, Warning, TEXT("HandleUnreliableRPC: Could not find target object: %s, skipping rpc at index: %d"), *ObjectRef.ToString(), Payload.Index);
+			UE_LOG(LogSpatialReceiver, Warning, TEXT("HandleRPC: Could not find target object: %s, skipping rpc at index: %d"), *ObjectRef.ToString(), Payload.Index);
 			continue;
 		}
 
 		const FClassInfo& ClassInfo = ClassInfoManager->GetOrCreateClassInfoByObject(TargetObject);
 
 		UFunction* Function = ClassInfo.RPCs[Payload.Index];
-		ApplyRPC(TargetObject, Function, Payload, FString());
+
+		FPendingRPCParamsPtr Params = MakeShared<FPendingRPCParams>(TargetObject, Function, MoveTemp(Payload));
+
+		// Apply if possible, queue otherwise
+		const FRPCInfo* RPCInfo = NetDriver->GetRPCInfo(TargetObject, Params->Function);
```

Comment:
Could you make GetRPCInfo return a const Ref so the check below is unneeded?

Reply:
What if for some reason GetRPCInfo function fails though? It only happened once in my experience - when the schema was invalid.

Reply:
If it fails then you crash. We don't support handling invalid schema.

Reply:
I think even returning a copy is fine since it's only 64 bit value.

Reply:
Changed to FRPCInfo.